src/Expr.cc

```diff
@@ -2625,10 +2625,7 @@ ValPtr AssignExpr::Eval(Frame* f) const
 TypePtr AssignExpr::InitType() const
 	{
 	if ( op1->Tag() != EXPR_LIST )
-		{
-		Error("bad initializer");
-		return nullptr;
-		}
+		Internal("bad initializer, first operand should be a list");
```

Comment:
Why would this be an internal error? Aren't we talking about something that's triggered by incorrect script code?

Reply:
I switched it back to being an `Error` and had `TableConstructorExpr::TableConstructorExpr` bail out if the result from initializing the type was null.